File: typescript/day_9/part_1.ts
import { readFileSync } from 'fs';

function solve(input: string): number {
    const lines: string[] = [];
    for (const line of input.trim().split('\n')) {
        const trimmed = line.trim();
        if (trimmed) {
            lines.push(trimmed);
        }
    }

    const tiles: [number, number][] = [];
    for (const line of lines) {
        const parts = line.split(',');
        const x = parseInt(parts[0]);
        const y = parseInt(parts[1]);
        tiles.push([x, y]);
    }

    const n = tiles.length;
    let max_area = 0;

    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            const [x1, y1] = tiles[i];
            const [x2, y2] = tiles[j];

            const width = Math.abs(x2 - x1) + 1;
            const height = Math.abs(y2 - y1) + 1;
            const area = width * height;

            max_area = Math.max(max_area, area);
        }
    }

    return max_area;
}

const input = readFileSync('../inputs/day_9.txt', 'utf-8');
const result = solve(input);
console.log(result);

File: java/day_9/Part1.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;

public class Part1 {
    public static int solve(String input) {
        List<String> lines = new ArrayList<>();
        for (String line : input.trim().split("\n")) {
            line = line.trim();
            if (!line.isEmpty()) {
                lines.add(line);
            }
        }

        List<int[]> tiles = new ArrayList<>();
        for (String line : lines) {
            String[] parts = line.split(",");
            int x = Integer.parseInt(parts[0]);
            int y = Integer.parseInt(parts[1]);
            tiles.add(new int[]{x, y});
        }

        int n = tiles.size();
        int max_area = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int x1 = tiles.get(i)[0], y1 = tiles.get(i)[1];
                int x2 = tiles.get(j)[0], y2 = tiles.get(j)[1];

                int width = Math.abs(x2 - x1) + 1;
                int height = Math.abs(y2 - y1) + 1;
                int area = width * height;

                max_area = Math.max(max_area, area);
            }
        }

        return max_area;
    }

    public static void main(String[] args) throws IOException {
        String input = new String(Files.readAllBytes(Paths.get("../inputs/day_9.txt")));
        int result = solve(input);
        System.out.println(result);
    }
}
